presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1976,7 +1981,7 @@ private HiveCompressionCodec getHiveCompressionCodec(ConnectorSession session, b
                         getTargetFileNames(partitionUpdate.getFileWriteInfos()),
                         partitionStatistics);
             }
-            else if (partitionUpdate.getUpdateMode() == APPEND) {
+            else if (!forceOverwrite && partitionUpdate.getUpdateMode() == APPEND) {
```

Comment:
The forceOverwrite flag seems hacky to me. We should think about how the partition mode can be of type overwrite.
In the HiveWriteFactory, while creating write parameters, we can check if the table is of type materialized view then we can set the partition update mode as overwrite instead of append. That would allow us to get rid of forceOverwrite flag.

Reply:
Yeah, I will take the good suggestion as we discussed and set the partition update mode from the ever beginning in HiveWriteFactory.